lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,56 @@
+module PulpProxy
+  class DiskUsage
+    include ::Proxy::Util
+    SIZE = { :kilobyte => 1_024, :megabyte => 1_048_576, :gigabyte => 1_073_741_824, :terabyte => 1_099_511_627_776 }
+
+    attr_reader :path, :stat, :size
+
+    def initialize(opts ={})
+      raise(::Proxy::Error::ConfigurationError, 'Unable to continue - must provide a path.') if opts[:path].nil?
+      @paths_hash = path_hash(opts[:path])
+      @path = @paths_hash.values
+      @size = SIZE[opts[:size]] || SIZE[:gigabyte]
+      @stat = {}
+      find_df
+      get_stat
+    end
+
+    def to_json
+      stat.to_json
+    end
+
+    private
+
+    attr_reader :command_path
+
+    def find_df
+      @command_path = which('df') || raise(::Proxy::Error::ConfigurationError, 'df command was not found unable to retrieve usage information.')
+    end
+
+    def command
+      [command_path, "-B", "#{size}", *path]
+    end
+
+    def get_stat
+      raw = Open3::popen3({"LC_ALL" => "C"}, *command) do |stdin, stdout, stderr, thread|
+        raise(::Proxy::Error::ConfigurationError, "#{command_path} raised an error: #{stderr}") unless stderr.read.empty?
```

Comment:
This can be multiline output, please join it with space. Also you could send it to logger.error as well, because this way it can be lost somewhere in Foreman.